src/main/java/online/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,153 @@
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getOrderFromResultSet(resultSet, connection));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order by id : " + id, e);
+        }
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM orders WHERE deleted = false";
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                orders.add(getOrderFromResultSet(resultSet, connection));
+            }
+            return orders;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all orders!", e);
+        }
+    }
+
+    @Override
+    public Order update(Order order) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "DELETE FROM orders WHERE order_id = ?";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, order.getId());
+            statement.executeUpdate();
+            addProductsToOrder(order, connection);
+            return order;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update order: " + order, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "UPDATE orders SET deleted = true WHERE order_id = ? AND deleted = false";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return true;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete order by id: " + id, e);
+        }
+    }
+
+    private Order getOrderFromResultSet(ResultSet resultSet, Connection connection) throws SQLException {
+        Long orderId = resultSet.getLong("order_id");
+        Long userId = resultSet.getLong("user_id");
+        Order order = new Order(userId);
+        order.setId(orderId);
+        order.setProducts(getOrderProduct(orderId, connection));
+        return order;
+    }
+
+    private List<Product> getOrderProduct(Long id, Connection connection) throws SQLException {
+        List<Product> products = new ArrayList<>();
+        String query = "SELECT * FROM products "
+                + "INNER JOIN orders_products "
+                + "ON orders_products.product_id = products.product_id "
+                + "WHERE orders_products.order_id = ?";
```

Comment:
подібні запити можна спрощувати:
```suggestion
        String query = "SELECT * FROM products p "
                + "INNER JOIN orders_products op "
                + "ON op.product_id = p.product_id "
                + "WHERE op.order_id = ?";
```